doc/api/net.md

```diff
@@ -117,49 +126,91 @@ when sockets were sent to forks.
 
 Callback should take two arguments `err` and `count`.
 
-### server.listen(handle[, backlog][, callback])
+### server.listen()
+
+Start a server listening for connections. A `net.Server` can be a TCP or
+local (domain sockets on UNIX, named pipes on Windows) server depending on
+what it listens to.
+
+*Note*: Unix named pipes (FIFOs) are not supported.
+
+Possible signatures:
+
+* [`server.listen(handle[, backlog][, callback])`][`server.listen(handle, backlog, callback)`]
+* [`server.listen(options[, callback])`][`server.listen(options, callback)`]
+* [`server.listen(path[, backlog][, callback])`][`server.listen(path, backlog, callback)`] for local servers
+* [`server.listen([port][, host][, backlog][, callback])`][`server.listen(port, host, backlog, callback)`] for TCP servers
+
+This function is asynchronous.  When the server starts listening, the
+[`'listening'`][] event will be emitted.  The last parameter `callback`
+will be added as a listener for the [`'listening'`][] event.
+
+All `listen()` methods can take a `backlog` parameter to specify the maximum
+length of the queue of pending connections. The actual length will be determined
+by the OS through sysctl settings such as `tcp_max_syn_backlog` and `somaxconn`
+on Linux. The default value of this parameter is 511 (not 512).
+
+
+Note:
+
+* All [`net.Socket`][] are set to `SO_REUSEADDR`.
```

Comment:
I trried http://man7.org/linux/man-pages/man7/socket.7.html but couldn't find an ID in that page(and it's a really long page)...maybe just add it because it's better than nothing? :3